opensrp-app/src/main/java/org/smartregister/repository/EventClientRepository.java

```diff
@@ -553,6 +560,51 @@ private boolean processEventClientCursor(List<EventClient> list, Cursor cursor)
                 new String[]{String.valueOf(startServerVersion), String.valueOf(lastServerVersion)});
     }
 
+    public P2pProcessRecordsService.EventClientQueryResult fetchEventClientsByRowId(long lastProcessedRowId) {
+        List<EventClient> list = new ArrayList<>();
+        Cursor cursor = null;
+        int maxRowId = 0;
+        try {
+            cursor = getWritableDatabase().rawQuery("SELECT " + ROWID + ",json FROM "
+                            + Table.event.name()
+                            + " WHERE "
+                            + ROWID
+                            + " > ?"
+                            + "ORDER BY " + ROWID + " ASC LIMIT 100",
```

Comment:
Shouldn't limit be the batchSize config

Reply:
Yes, it should. I'll change that

Reply:
So I am wrong above **UP**. That method is used to process the event clients and not to fetch for transfer. I noticed that if it's too big, then the processing becomes slow and a lot of GCs due to the JSONObject and further processing that creates required objects